api/client/client.go

```diff
@@ -25,38 +25,25 @@ import (
 	"github.com/docker/compose-cli/api/containers"
 	apicontext "github.com/docker/compose-cli/api/context"
 	"github.com/docker/compose-cli/api/context/store"
+	"github.com/docker/compose-cli/api/errdefs"
 	"github.com/docker/compose-cli/api/resources"
 	"github.com/docker/compose-cli/api/secrets"
 	"github.com/docker/compose-cli/api/volumes"
 )
 
 // New returns a backend client associated with current context
 func New(ctx context.Context) (*Client, error) {
-	return newWithDefaultBackend(ctx, "")
-}
-
-// NewWithDefaultLocalBackend returns a backend client associated with current context or local backend if on default context type
-func NewWithDefaultLocalBackend(ctx context.Context) (*Client, error) {
-	return newWithDefaultBackend(ctx, store.LocalContextType)
-}
-
-func newWithDefaultBackend(ctx context.Context, defaultBackend string) (*Client, error) {
-	currentContext := apicontext.CurrentContext(ctx)
-	s := store.ContextStore(ctx)
+	currentContext := apicontext.Current()
```

Comment:
This method centralises checks on singleton to ensure all is initialised ; need to also check if currentContext and store are properly initlised, and return clear errors if not, as with backend